### ISSUE TITLE
dSYMs success, but fails?

### ISSUE THREAD

**Issue by MaximusMcCann** [getsentry/fastlane-plugin-sentry#2]
I see it saying `dSYM successfully uploaded to Sentry!`, but then it fails.  Not sure what's going on.
Appears it did succeed, looking at the Sentry website.

shortened Fastfile:

```
platform :ios do
  #ENV["SENTRY_AUTH_TOKEN"] = in env profile
  ENV["SENTRY_ORG_SLUG"] = "<our-slug>"
  ENV["SENTRY_PROJECT_SLUG"] = "ios"
end
...
...
private_lane :__upload_dsyms do
  SCHEME = ENV["BUILD_SCHEME"]
  BUILD_NUMBER = __build_number
  
  download_dsyms(build_number: BUILD_NUMBER, app_identifier: ENV["APP_IDENTIFIER"])
  upload_symbols_to_crashlytics
  sentry_upload_dsym(dsym_path: "./#{SCHEME}.app.dSYM.zip")
end
```

Output:

```
[16:15:12]: Successfully uploaded dSYM files to Crashlytics 💯
[16:15:12]: --------------------------------
[16:15:12]: --- Step: sentry_upload_dsym ---
[16:15:12]: --------------------------------
[16:15:12]: Will upload dSYM(s) to https://app.getsentry.com/api/0/projects/aaa/ios/files/dsyms/
[16:15:12]: Uploading... /Users/distiller/a41.dSYM.zip
[16:15:51]: dSYM successfully uploaded to Sentry!
[16:15:51]: Uploading... ./aaa.app.dSYM.zip
[16:16:52]: Variable Dump:
[16:16:52]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>nil, :LANE_NAME=>"devTestFlightPush", :FL_CHANGELOG=>"aoeute\n", :GET_IPA_INFO_PLIST_VALUE_CUSTOM_VALUE=>"981.1481240441", :DSYM_PATHS=>["/Users/distiller/aa.dev-2.2.0-981.1481240441.dSYM.zip"]}
[16:16:52]: Error while trying to upload dSYM to Sentry
```

**Comment by joshdholtz** [comment]
@MaximusMcCann Thanks for letting me know! I will look into this and also print out more useful information from the plugin if/when it fails

**Comment by MaximusMcCann** [comment]
Cheers on the better logging.  Now I see: `Error: Request Timeout`

```
[08:52:55]: Successfully uploaded dSYM files to Crashlytics 💯
[08:52:55]: --------------------------------
[08:52:55]: --- Step: sentry_upload_dsym ---
[08:52:55]: --------------------------------
[08:52:55]: Will upload dSYM(s) to https://app.getsentry.com/api/0/projects/aaa/ios/files/dsyms/
[08:52:55]: Uploading... /Users/distiller/ios/aaa.dSYM.zip
[08:53:33]: dSYM successfully uploaded to Sentry!
[08:53:33]: Uploading... /Users/distiller/ios/aaa.app.dSYM.zip
[08:54:42]: Error: Request Timeout
```

**Comment by joshdholtz** [comment]
@MaximusMcCann I am not able to duplicate this on my end. Would you be able to email me your `dSYM.zip` so I can try to debug what is wrong?

**Comment by MaximusMcCann** [comment]
I see the dSYM it has troubles with is the dSYM that contains all my Pods.  The dSYM for our app works fine.  More output:


```
[14:22:48]: Looking for dSYM files for com.aa.name (1019)
[14:23:12]: 🔑  Successfully downloaded dSYM file for 2.3.0 - 1019 to 'com.aa.name-2.3.0-1019.dSYM.zip'
[14:23:16]: -------------------------------------------
[14:23:16]: --- Step: upload_symbols_to_crashlytics ---
[14:23:16]: -------------------------------------------
[14:23:16]: Extracting '/Users/distiller/ios/client.app.dSYM.zip'...
[14:23:16]: $ unzip -qo /Users/distiller/ios/client.app.dSYM.zip
[14:23:17]: Uploading 'Alamofire.framework.dSYM'...
[14:23:18]: Uploading 'Async.framework.dSYM'...
[14:23:19]: Uploading 'Bolts.framework.dSYM'...
.
.
.
[14:23:52]: Uploading 'zxcvbn_ios.framework.dSYM'...
[14:23:52]: Extracting '/Users/distiller/ios/com.aa.name-2.3.0-1019.dSYM.zip'...
[14:23:52]: $ unzip -qo /Users/distiller/ios/com.aa.name-2.3.0-1019.dSYM.zip
[14:23:53]: Uploading '03FCE59E-9731-35AB-AEFA-096E5667D2DD.dSYM'...
[14:23:53]: Uploading '05B7BB16-9CD6-3601-A313-A61BAB071E2F.dSYM'...
.
.
.
[14:24:44]: Uploading 'FE11C790-4E46-3E2E-A3D3-8DB2EB0AC49F.dSYM'...
[14:24:44]: Successfully uploaded dSYM files to Crashlytics 💯
[14:24:44]: --------------------------------
[14:24:44]: --- Step: sentry_upload_dsym ---
[14:24:44]: --------------------------------
[14:24:44]: Will upload dSYM(s) to https://app.getsentry.com/api/0/projects/aaa/ios/files/dsyms/
[14:24:44]: Uploading... /Users/distiller/ios/com.aa.name-2.3.0-1019.dSYM.zip
[14:25:14]: dSYM successfully uploaded to Sentry!
[14:25:14]: Uploading... /Users/distiller/ios/name.app.dSYM.zip
[14:26:18]: Error: Request Timeout
```

**Comment by joshdholtz** [comment]
@MaximusMcCann I just got this with a dsym for one of my apps so I should be able to debug this easier now 😊

**Comment by joshdholtz** [comment]
@MaximusMcCann Can you test something out for me? 😊 

What I did was add add an ability to configure a timeout duration (which may or may not work with `rest-client` which uploads the dSYM) so I also provided a way which shells out the upload process to `curl` which the timeout feature definitely works on.

Let me know if you have any questions/issues with this! Hoping this works out for you 🚀 

1) Change your `Pluginfile` to this... (this will install my experimental branch for you)
```rb
#gem 'fastlane-plugin-sentry'
gem 'fastlane-plugin-sentry', git: 'https://github.com/getsentry/fastlane-plugin-sentry.git', branch: 'timeout-and-curl'
```
2) Set new `timeout` and `use_curl` config items in either your `Fastfile` or `.env` file
```rb
# Your call to sentry should look something like this...
sentry_upload_dsym(
    timeout: 120,
    use_curl: true
  )
```

```
// This is what could/should be added to your .env file
SENTRY_USE_CURL=true
SENTRY_TIMEOUT=180
```

**Comment by joshdholtz** [comment]
@MaximusMcCann This has been released in version `0.1.2`. Can you give this a try at your earliest convenience and let me know **when** it works for you 😇

**Comment by MaximusMcCann** [comment]
@joshdholtz success!  Thank you!
